src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/PhysicalPlanFactory.java

```diff
@@ -193,51 +193,24 @@ public static PhysicalPlan createPlan( final UnaryPhysicalOp rootOp,
 
 	/**
 	 * Creates a physical plan with a bgpAdd as root operator.
-	 * The physical algorithm of bgpAdd is determined by the type of given physical operator.
-	 * The given subplan becomes the children of the root operator.
-	 */
-	public static PhysicalPlan createPlanBasedOnTypeOfGivenPhysicalOp( final PhysicalOperator op, final LogicalOpBGPAdd lop, final PhysicalPlan subplan ) {
-		if ( op instanceof PhysicalOpIndexNestedLoopsJoin) {
-			return createPlanWithIndexNLJ( lop, subplan );
-		}
-		else if ( op instanceof PhysicalOpBindJoinWithFILTER) {
-			return createPlanWithBindJoinFILTER( lop, subplan );
-		}
-		else if ( op instanceof PhysicalOpBindJoinWithUNION) {
-			return createPlanWithBindJoinUNION( lop, subplan );
-		}
-		else if ( op instanceof PhysicalOpBindJoinWithVALUES) {
-			return createPlanWithBindJoinVALUES( lop, subplan );
-		}
-		else {
-			return createPlan( lop, subplan );
+	 * The root operator uses the same physical algorithm as the given physical operator, which is usually a specific physical operator for bgpAdd.
+	 * The given subplan becomes the single child of the root operator.
+	 */
+	public static PhysicalPlan createPlanBasedOnTypeOfGivenPhysicalOp( final LogicalOpBGPAdd lop, final Class<? extends PhysicalOperator> opClass, final PhysicalPlan subplan ) {
+		switch ( opClass.getName() ) {
+			case "PhysicalOpIndexNestedLoopsJoin":
```

Comment:
This may not work correctly in all cases; in particular, it doesn't cover cases in which `opClass` is a subclass of any of the classes recognized here (such as `PhysicalOpIndexNestedLoopsJoin`). To make this work for subclasses as well, I think you can use the `isAssignableFrom` method of `Class`-- try something along the following lines:
```java
if ( PhysicalOpIndexNestedLoopsJoin.class.isAssignableFrom(opClass) ) {
    return createPlanWithIndexNLJ( lop, subplan );
}
else if ...
```

...but please double check whether it really works that way.

Reply:
Thanks, I have updated this method using the approach you proposed. 
In addition, I created a unit test for rule 'RuleMergeTPAddOfBGPAddIntoBGPAdd' to check whether this method works as expected.